app/web/pages/messages/[[...slug]].tsx

```diff
@@ -0,0 +1,62 @@
+import { appGetLayout } from "components/AppRoute";
+import GroupChatsTab from "features/messages/groupchats/GroupChatsTab";
+import GroupChatView from "features/messages/groupchats/GroupChatView";
+import MessagesHeader from "features/messages/MessagesHeader";
+import HostRequestView from "features/messages/requests/HostRequestView";
+import RequestsTab from "features/messages/requests/RequestsTab";
+import NotFoundPage from "features/NotFoundPage";
+import { GetStaticPaths, GetStaticProps } from "next";
+import { useRouter } from "next/router";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { ReactNode } from "react";
+import { messageTypeStrings } from "routes";
+
+export const getStaticPaths: GetStaticPaths = () => ({
+  paths: [],
+  fallback: "blocking",
+});
+
+export const getStaticProps: GetStaticProps = async ({ locale }) => ({
+  props: {
+    ...(await serverSideTranslations(locale ?? "en", ["global", "messages"])),
+  },
+});
+export default function LeaveReferencePage() {
+  const router = useRouter();
+  const slugs =
+    typeof router.query.slug === "undefined"
+      ? ["chats"]
+      : typeof router.query.slug === "string"
+      ? [router.query.slug]
+      : router.query.slug;
+  let content: ReactNode;
+
+  const tab = messageTypeStrings.find((valid) => valid === slugs?.[0]);
+
+  if (slugs[0] === "chats") {
+    const id = Number.parseInt(slugs?.[1]);
+    if (isNaN(id)) {
+      content = <GroupChatsTab />;
+    } else {
+      content = <GroupChatView chatId={id} />;
+    }
+  } else if (slugs[0] === "request") {
+    const id = Number.parseInt(slugs?.[1]);
+    if (isNaN(id)) {
+      return <NotFoundPage />;
+    } else {
+      content = <HostRequestView hostRequestId={id} />;
```

Comment:
In the old switch logic, the `HostRequestView` doesn't have the `MessageHeader` rendered (probably why `header` is horribly repeated, I kinda like this a bit better lol). Do we now want it instead?

Reply:
Does it look odd? It's easier to get around.

Reply:
Had another look at this and I wonder if that was why we hid the tab bar in some places, as I am getting this error/warning in the console now:
![Screenshot 2022-01-05 001640](https://user-images.githubusercontent.com/13669362/148141178-e7436a98-b4df-4d60-b777-cfcba3381600.png)

Problem seems to be that when we click requests from either the surfing or hosting tab, it takes the user to `/messages/request/[request-id]` and so we have no way of knowing whether "request" should activate the "hosting" or "surfing" tab...